front/lib/swr.ts

```diff
@@ -1028,6 +1042,83 @@ export function useWorkspaceEnterpriseConnection({
   };
 }
 
+type UseDataSourceKey = {
+  workspaceSid: string;
+  dataSourceName: string;
+  internalIds: string[];
+};
+
+export function useDataSourceNodes(
+  { workspaceSid, dataSourceName, internalIds }: UseDataSourceKey,
```

Comment:
On the front-end side, we always assume that `workspaceId` is actually the `sId` 🙃.

```suggestion
  { workspaceid, dataSourceName, internalIds }: UseDataSourceKey,
```